models/issue.go

```diff
@@ -1277,13 +1277,24 @@ func GetParticipantsIDsByIssueID(issueID int64) ([]int64, error) {
 
 // GetParticipantsByIssueID returns all users who are participated in comments of an issue.
 func GetParticipantsByIssueID(issueID int64) ([]*User, error) {
-	return getParticipantsByIssueID(x, issueID)
+	issue, err := getIssueByID(x, issueID)
+	if err != nil {
+		return nil, err
+	}
+	userIDs, err := getParticipantsByIssueID(x, issue)
+	if err != nil {
+		return nil, err
+	}
+	if len(userIDs) == 0 {
+		return nil, nil
+	}
+	return GetUsersByIDs(userIDs)
 }
 
-func getParticipantsByIssueID(e Engine, issueID int64) ([]*User, error) {
+func getParticipantsByIssueID(e Engine, issue *Issue) ([]int64, error) {
```

Comment:
Perhaps... ?
```suggestion
func (issue *Issue) getActiveCommentersIDs(e Engine) ([]int64, error) {
```
In any case, it's not "ByIssueID" anymore.

The problem is that there is also a `GetParticipantsIDsByIssueID` function that does the roughly same -except the availability filtering- but _doesn't_ add the original poster. It would be better to differentiate them with distinctive names.

Reply:
@guillep2k I think getParticipants is the better name because the issueCreateor is als a participanct ... yes before my change the better name would be something like GetActiveCommenters or so